### PR TITLE
Improve UX of unavailable ledger balance

### LINKED ISSUE #3778: Feeling of panic immidiately on every start
1. Leave about:preferences#payments open
2. Restart the browser and check the Payments page

Expected: account balance to display a cached value from the previous session until Brave has updated the currency exchange rates, etc.

Actual: account balance “0 USD” displayed for a few seconds and my heart skips a beat or two and I start thinking about whether I’ve backed up my Brave user data or not. After some seconds the correct value is displayed.

### PR DESCRIPTION
**Pull request by ayumi**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Fix #3778 #3785

Adds `ledgerInfo.error` for debugging and UI.

Auditors: @mrose17 @bbondy 

Test Plan:

Offline case:
1. Disable internet
2. Re/open Brave
3. Open Preferences -> Payments
4. Observe that account balance shows "error, can't retrieve data" and not 0

Online case:
1. In Brave, open Preferences -> Payments and leave it open
2. Close and reopen Brave
4. Observe Payments balance shows "loading" or the correct amount

Offline case / Expected result:
screen shot 2016-09-07 at 18 54 33

### CODE REVIEW

File: app/ledger.js [hunk 1]
```diff
@@ -744,7 +744,14 @@ var ledgerInfo = {
 
   hasBitcoinHandler: false,
 
-  _internal: {}
+  _internal: {},
+
+  // if the last updateLedgerInfo happened concurrently with an error, this
+  // saves it for UI or debugging
+  error: null // {
```
Comment (mrose17) [inline app/ledger.js]: as @diracdeltas is fond of saying: _don't forget to update docs/state.md_

**Comment by mrose17** [comment]
@ayumi - have you been able to reproduce either of the issues? i never have.

my theory has been that `(! this.props.ledgerData) === true`, but i until i can reproduce it, i can't really say.

on a mac, sometimes i see the payments panel show the 'Welcome to Brave Payments' page and then flip into the actual page a few 100ms later. 

on linux, i suspect that something similar may be going on.

what's your thinking? thanks!

File: app/ledger.js (hunk 1 above)
Comment (ayumi) [inline app/ledger.js]: thx, that is a good idea

**Comment by ayumi** [comment]
@mrose17 I can repro #3785 and #3778 at https://github.com/brave/browser-laptop/commit/5b10525e191f2d74c8241e7b585a9574fb25d453 on Mac OS X.

I've repro'd by disabling wifi and by specifically blocking Brave using [Little Snitch](https://www.obdev.at/products/littlesnitch/index.html).

This PR fixes it for me.

**Comment by mrose17** [comment]
@ayumi - excellent!

please merge the PR into master!

**Pull request merged**